src/Extensions/AzureBlobPayloads/PayloadStore/BlobPayloadStore.cs

```diff
@@ -0,0 +1,215 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System.Globalization;
+using System.IO.Compression;
+using System.Text;
+using Azure;
+using Azure.Core;
+using Azure.Storage.Blobs;
+using Azure.Storage.Blobs.Models;
+using Microsoft.DurableTask.Converters;
+
+namespace Microsoft.DurableTask;
+
+/// <summary>
+/// Azure Blob Storage implementation of <see cref="IPayloadStore"/>.
+/// Stores payloads as blobs and returns opaque tokens in the form "blob:v1:&lt;container&gt;:&lt;blobName&gt;".
+/// </summary>
+internal sealed class BlobPayloadStore : IPayloadStore
+{
+    const string TokenPrefix = "blob:v1:";
+    const string ContentEncodingGzip = "gzip";
+    const int DefaultCopyBufferSize = 81920;
+    const int MaxRetryAttempts = 8;
+    const int BaseDelayMs = 250;
+    const int MaxDelayMs = 10_000;
+    const int MaxJitterMs = 100;
+    const int NetworkTimeoutMinutes = 2;
+
+    // Jitter RNG for retry backoff
+    static readonly object RandomLock = new object();
+    static readonly Random SharedRandom = new Random();
+    readonly BlobContainerClient containerClient;
+    readonly LargePayloadStorageOptions options;
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="BlobPayloadStore"/> class.
+    /// </summary>
+    /// <param name="options">The options for the blob payload store.</param>
+    /// <exception cref="ArgumentNullException">Thrown when <paramref name="options"/> is null.</exception>
+    /// <exception cref="ArgumentException">Thrown when <paramref name="options.ConnectionString"/> is null or empty.</exception>
+    public BlobPayloadStore(LargePayloadStorageOptions options)
```

Comment:
Same comment here, I think we should expose identity connections unless you can specify that in the connection string.